packages/@react-spectrum/s2/src/Avatar.tsx

```diff
@@ -53,13 +53,17 @@ const imageStyles = style({
   }
 }, getAllowedOverrides({width: false}));
 
-export const AvatarContext = createContext<ContextValue<Partial<AvatarProps>, DOMRefValue<HTMLImageElement>>>(null);
+export interface AvatarContextValue extends AvatarProps {
+  render?: (avatar: ReactNode) => ReactNode
```

Comment:
I think rather than adding this prop we could just make Avatar have CenterBaseline by default. I can't really think of a case where you'd want to align text with it at the bottom. We may be able to avoid the extra wrapper div as well since Image already has a wrapper div. We could potentially put the `::before` directly on that.